### PR TITLE
Move chart to charts directory and add workflow to publish new chart versions

### PR DESCRIPTION
**Pull request by krmichel**
**Is this a bug fix or adding new feature?**
This is a feature
**What is this PR about? / Why do we need it?**
Currently managing the index.yaml of the helm chart repository is a manual process and gets out of date quickly.  This will move the chart to be a sub-directory of charts which is a pretty standard structure.  It also adds a workflow action that will publish a release of the chart and update the index.yaml in the chart repo.  The workflow action will also fail if the chart is updated and merged to master without the version being updated (the version of the chart should be increased whenever the chart is modified). 
**What testing is done?** 
I tried out the workflow action in my own repo

### CODE REVIEW

**Comment by krmichel** [comment]
@leakingtapan This should not be merged until after PR #623 has been merged and the index.yaml is showing as updated in the chart repository at https://kubernetes-sigs.github.io/aws-ebs-csi-driver/index.yaml to show all the available versions or the workflow action won't have the correct version of the index to merge the new version into.

**Comment by krmichel** [comment]
This should prevent issues like #557 #587 #617 in the future

**Comment by krmichel** [comment]
@jsafrane @d-nishi @leakingtapan Can I get an okay to test?

**Comment by krmichel** [comment]
@jsafrane @d-nishi @leakingtapan Tests have all passed when you get a chance to look at this.  Please look at PR #623 first though.

**Comment by krmichel** [comment]
@wongma7 Here is the PR I mentioned earlier so you can verify the kube version change

File: .github/workflows/helm-chart-release.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@v1.1.0
```
Comment (wongma7) [inline .github/workflows/helm-chart-release.yaml:23]: cool. ref for me https://github.com/helm/chart-releaser-action

File: docs/README.md [hunk 2]
```diff
@@ -119,13 +119,21 @@ kubectl get pods -n kube-system
...
+
+Then install a release of the driver using the chart
 ```sh
 helm upgrade --install aws-ebs-csi-driver \
     --namespace kube-system \
     --set enableVolumeScheduling=true \
     --set enableVolumeResizing=true \
     --set enableVolumeSnapshot=true \
-    https://github.com/kubernetes-sigs/aws-ebs-csi-driver/releases/download/v0.7.1/helm-chart.tgz
+    aws-ebs-csi-driver/aws-ebs-csi-driver
```
Comment (wongma7) [inline docs/README.md:136]: could you explain this syntax, having trouble understanding it. I guess the chart name is aws-ebs-csi-driver but what does it mean to have aws-ebs-csi-driver/aws-ebs-csi-driver?

 helm upgrade [RELEASE] [CHART] [flags]

Comment (wongma7) [reply docs/README.md:136]: Never mind, I've got it. it is a chart reference. A/B where A is the repo name and B is the chart name.

https://helm.sh/docs/helm/helm_install/

**Comment by wongma7** [comment]
Sorry do you mind rebasing I merged a months old small pr fixing some naming issues. thanks

**Comment by krmichel** [comment]
@wongma7 I have resolved the conflict

**Comment by wongma7** [comment]
/lgtm
/approve

thank you very much for all your contributions to the helm chart!!!

**Comment by krmichel** [comment]
@wongma7 it looks like the gh-pages branch is protected and declined the index.yaml update from the action.
https://github.com/kubernetes-sigs/aws-ebs-csi-driver/runs/1513656332
It look like I might be able to have it create PRs, but I think it would be better if it could be setup to allow the publish without a PR so there aren't move of the chart repo is out of date issues.  Thoughts?

**Comment by wongma7** [comment]
let me check if I can turn off branch protection. https://github.com/kubernetes/test-infra/blob/2abb7ff26579325f6a335990003f665755c96d7a/prow/cmd/branchprotector/README.md#advanced-configuration

**Pull request merged**